src/prop/minisat/core/Solver.cc

```diff
@@ -51,10 +51,10 @@ namespace {
  * resolution proofs get removed when popping the context but the SAT solver
  * keeps using them.
  */
-bool assertionLevelOnly()
+bool assertionLevelOnly(const Options& opts)
 {
-  return (options::produceProofs() || options::unsatCores())
-         && options::incrementalSolving();
+  return (opts.smt.produceProofs || opts.smt.unsatCores)
+         && opts.base.incrementalSolving;
 }
```

Comment:
This looks like this should be a flag that is set once instead of always checking the same options over again.